pkg/expression/contextstatic/exprctx_test.go

```diff
@@ -145,16 +147,47 @@ func TestExprCtxColumnIDAllocator(t *testing.T) {
 	require.Equal(t, int64(3), ctx.AllocPlanColumnID())
 
 	// Apply with new allocator
-	alloc = context.NewSimplePlanColumnIDAllocator(1024)
+	alloc = exprctx.NewSimplePlanColumnIDAllocator(1024)
 	ctx3 := ctx.Apply(WithColumnIDAllocator(alloc))
 	require.Same(t, alloc, ctx3.columnIDAllocator)
 	require.NotSame(t, ctx.columnIDAllocator, ctx3.columnIDAllocator)
 	require.Equal(t, int64(1025), ctx3.AllocPlanColumnID())
 	require.Equal(t, int64(4), ctx.AllocPlanColumnID())
 
 	// New context with allocator
-	alloc = context.NewSimplePlanColumnIDAllocator(2048)
+	alloc = exprctx.NewSimplePlanColumnIDAllocator(2048)
 	ctx4 := NewStaticExprContext(WithColumnIDAllocator(alloc))
 	require.Same(t, alloc, ctx4.columnIDAllocator)
 	require.Equal(t, int64(2049), ctx4.AllocPlanColumnID())
 }
+
+func TestMakeExprContextStatic(t *testing.T) {
+	evalCtx := NewStaticEvalContext()
+	planCacheTracker := contextutil.NewPlanCacheTracker(evalCtx.warnHandler)
+	obj := NewStaticExprContext(
+		WithEvalCtx(evalCtx),
+		WithCharset("a", "b"),
+		WithDefaultCollationForUTF8MB4("c"),
+		WithBlockEncryptionMode("d"),
+		WithSysDateIsNow(true),
+		WithNoopFuncsMode(1),
+		WithRng(mathutil.NewWithSeed(12345678)),
+		WithPlanCacheTracker(&planCacheTracker),
+		WithColumnIDAllocator(exprctx.NewSimplePlanColumnIDAllocator(1)),
+		WithConnectionID(1),
+		WithWindowingUseHighPrecision(true),
+		WithGroupConcatMaxLen(1),
+	)
+
+	ignorePath := []string{
+		"^\\$\\.staticExprCtxState\\.evalCtx\\..*",
+		"^\\$\\.staticExprCtxState\\.planCacheTracker\\..*",
+		"^\\$\\.staticExprCtxState\\.columnIDAllocator\\..*",
+		"^\\$.*\\.mu",
+	}
+	h := deeptest.NewStaticTestHelper(ignorePath)
+	h.AssertNotEmpty(t, reflect.ValueOf(obj), "$")
```

Comment:
We should also check obj's fields is not the same value of `NewStaticExprContext()` without any options